src/server/handlers/consents/{ID}/revoke.ts

```
@@ -0,0 +1,99 @@
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ --------------
+ ******/
+import {
+  patchConsentRevoke,
+  isConsentRequestInitiatedByValidSource,
+  revokeConsentStatus
+} from '../../../../domain/consents/revoke'
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import Logger from '@mojaloop/central-services-logger'
+import { Enum } from '@mojaloop/central-services-shared'
+import { consentDB } from '../../../../lib/db'
+import { Consent } from '../../../../model/consent'
+
+export async function validateRequestAndRevokeConsent (
+  request: Request): Promise<void> {
+  const consentId = request.params.id
+
+  try {
+    // Fetch consent from database using ID
+    let consent: Consent
+    try {
+      consent = await consentDB.retrieve(consentId)
+    } catch (error) {
+      Logger.push(error)
+      Logger.error('Error in retrieving consent')
+
+      // If consent cannot be retrieved using given ID, send PUT ...error back
+      // TODO: Error Handling dealt with in future ticket #355
+      throw (new Error('NotImplementedYetError'))
+    }
+
+    // If request is not intiated by valid source, send PUT ...error back
+    if (!isConsentRequestInitiatedByValidSource(consent, request)) {
+      // TODO: Error Handling dealt with in future ticket #355
+      throw (new Error('NotImplementedYetError'))
+    }
+
+    // If Consent is ACTIVE, revoke it and update database
+    if (consent.status === 'ACTIVE') {
```

Maybe we could throw an error if the `status` is not active. Maybe `BAD REQUEST`?

---

do you mean response? cant do that because we sent back the ACCEPTED code already